appcues/src/main/java/com/appcues/AppcuesExperienceActionsImpl.kt

```diff
@@ -0,0 +1,54 @@
+package com.appcues
+
+import com.appcues.action.ActionProcessor
+import com.appcues.action.ExperienceAction
+import com.appcues.analytics.AnalyticsTracker
+import com.appcues.analytics.ExperienceLifecycleEvent.StepInteraction.InteractionType.BUTTON_TAPPED
+import com.appcues.data.model.RenderContext
+import com.appcues.data.model.StepReference.StepOffset
+import com.appcues.ui.ExperienceRenderer
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.launch
+
+// Suppressing until figure out a way to make this class more concise
```

Comment:
in other places we have just injected the scope and then had properties pulled out from the scope - just moving the code around though, no real difference.

Reply:
@andretortolano just noting another implementation idea from the iOS side (cc @mmaatttt ) - where it is just queueing up the various ExperienceActions (i.e. `AppcuesTrackAction`) 

https://github.com/appcues/appcues-ios-sdk/pull/573/files#diff-0d15d49ce022bd49e4902adc68df3c80a91afefc0800e31ce578c59f4a5924e0R30

maybe that is not needed here, but noting the idea. I was wondering if we needed the action queue capability there (the Channel) or a way to wait on these actions to complete - for instance, should `nextStep()` be a suspend function? I'm not sure. Are there scenarios where the calling component would need to know when the action completes to proceed to some additional logic? Since this is in the public API, would be good to include that now, if needed.